### PR TITLE
add searchbar

### LINKED ISSUE #217: A searchbox for making it easier to search skills.
**Is your feature request related to a problem? Please describe.**
There seems to be a lot of options in skills to choose from and manually searching and selecting them makes this a bit of a tedious task.

**Describe the solution you'd like**
I want a searchbox where I can easily filter through the Skills and select them. Makes the workflow a little convenient.

**Describe alternatives you've considered**
None

**Additional context**
Here's a basic implementation by me

Join the **Discord Server** for further discussions.

Server Link: https://discord.gg/HHMs7Eg

### PR DESCRIPTION
**Pull request by j23saw**
Closes #217 
Added a search bar to make it convenient to search for skills.
Here's a demo:
ezgif com-video-to-gif(1)

Check out the live demo here: https://festive-khorana-deef28.netlify.app/
Suggest some changes if needed.

### CODE REVIEW

**Comment by rahuldkjain** [comment]
@j23saw 
Few suggestions:
1. Change the border of `Search Skill` input box with respect to the design style guide. I mean the border and border-radius must be the same as the other buttons.
2. Border Shadow on the Skills categories can be removed. Although we can keep an `hr` tag below the category title.

**Comment by j23saw** [comment]
@rahuldkjain working on it!

**Comment by j23saw** [comment]
I have implemented your suggestions, any more changes needed?

File: package.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
     "gatsby-plugin-manifest": "^2.4.14",
     "gatsby-plugin-offline": "^3.2.13",
     "gatsby-plugin-react-helmet": "^3.3.6",
-    "gatsby-plugin-sharp": "^2.6.14",
+    "gatsby-plugin-sharp": "^2.6.38",
```
Comment (rahuldkjain) [inline package.json:15]: remove this change from PR.

File: package.json [hunk 2]
```diff
@@ -44,7 +44,7 @@
   "license": "0BSD",
   "scripts": {
     "build": "gatsby build",
-    "develop": "gatsby develop",
+    "develop": "gatsby develop -H 0.0.0.0",
```
Comment (rahuldkjain) [inline package.json:47]: remove this change too

**Review by rahuldkjain — CHANGES_REQUESTED** [review]
Please ignore the changes in `package-lock.json` in this PR. 
Fix the comments.

**Comment by j23saw** [comment]
@rahuldkjain please review the changes.

**Comment by rahuldkjain** [comment]
Will review soon @j23saw.

**Review by rahuldkjain — APPROVED** [review]
Superb @j23saw. Really liked the user experience of the search bar. Impressed to see that you took care of edge cases like `no results found` too.

**Comment by j23saw** [comment]
Thank you so much! Really appreciate your feedback✌️

**Pull request merged**